worker/worker.go

```diff
@@ -638,23 +650,23 @@ func (w *worker) objectsKeyHandlerPUT(jc jape.Context) {
 	}
 	w.pool.setCurrentHeight(up.CurrentHeight)
 	usedContracts := make(map[consensus.PublicKey]types.FileContractID)
-	for {
-		// fetch contracts
-		bcs, err := w.bus.ContractSetContracts(up.ContractSet)
-		if jc.Check("couldn't fetch contracts from bus", err) != nil {
-			return
-		}
 
-		r := io.LimitReader(jc.Request.Body, int64(up.MinShards)*rhpv2.SectorSize)
+	// fetch contracts
+	bcs, err := w.bus.ContractSetContracts(up.ContractSet)
```

Comment:
rereading my own PR I realise this might've been inside of the `for` loop on purpose... was that the case? You can argue that on upload we should re-fetch the contracts after every slab but it's debatable...